tests/test_socketdelay.js

```diff
@@ -56,7 +56,28 @@ describe('`socketDelay()`', () => {
     req.end()
   })
 
-  it('emits a timeout if not idle for long enough', done => {
+  it('emits a timeout - with options.timeout', done => {
+    nock('http://example.test')
+      .get('/')
+      .socketDelay(10000)
+      .reply(200, 'OK')
+
+    const onEnd = sinon.spy()
+
+    const req = http.request('http://example.test', { timeout: 5000 }, res => {
+      res.setEncoding('utf8')
```

Comment:
Does this `setEncoding` have an affect on the test?

Reply:
I don't know to be fully honest. I copied the test case from above and made the minimal number of changes to make it work for `options.timeout`. I can take a look into it and adjust it for both tests cases if needed?

Reply:
@mastermatt any thoughts?